crates/providers/blobstore-s3/src/client.rs

```diff
@@ -104,246 +121,89 @@ impl StorageClient {
+    pub async fn container_exists(
         &self,
-        _ctx: &Context,
-        bucket_id: &str,
-        object_id: &str,
-    ) -> Result<ObjectMetadata, RpcError> {
-        let bucket_id = self.unalias(bucket_id);
-        match self
-            .s3_client
-            .head_object()
-            .bucket(bucket_id)
-            .key(object_id)
-            .send()
-            .await
-        {
-            Ok(HeadObjectOutput {
-                last_modified,
-                content_length,
-                content_type,
-                content_encoding,
-                ..
-            }) => Ok(ObjectMetadata {
-                container_id: bucket_id.to_string(),
-                object_id: object_id.to_string(),
-                last_modified: to_timestamp(last_modified),
-                content_type,
-                content_encoding,
-                content_length: content_length as u64,
-            }),
-            Err(SdkError::ServiceError {
-                err:
-                    HeadObjectError {
-                        kind: HeadObjectErrorKind::NotFound(_),
-                        ..
-                    },
-                ..
-            }) => Err(RpcError::Other(format!(
-                "Not found: Bucket({}) Object({})",
-                bucket_id, object_id,
-            ))),
-            Err(e) => Err(RpcError::Other(format!(
-                "get_object_metadata for Bucket({}) Object({}): {}",
-                bucket_id, object_id, e
-            ))),
-        }
-    }
-
-    /// Sends bytes to actor in a single rpc message.
-    /// If successful, returns number of bytes sent (same as chunk.content_length)
-    #[instrument(level = "debug", skip(self, ctx, chunk), fields(actor_id = ?ctx.actor, object_id = %chunk.object_id, container_id = %self.unalias(&chunk.container_id)))]
-    async fn send_chunk(&self, ctx: &Context, mut chunk: Chunk) -> Result<u64, RpcError> {
-        chunk.container_id = self.unalias(&chunk.container_id).to_string();
-        let receiver = ChunkReceiverSender::for_actor(self.ld.as_ref());
-        if let Err(e) = receiver.receive_chunk(ctx, &chunk).await {
-            let err = format!(
-                "sending chunk error: Bucket({}) Object({}) to Actor({}): {}",
-                &chunk.container_id, &chunk.object_id, &self.ld.actor_id, e
-            );
-            error!(error = %e, "sending chunk error");
-            Err(RpcError::Rpc(err))
-        } else {
-            Ok(chunk.bytes.len() as u64)
+        ctx: &Context,
+        arg: &ContainerId,
+    ) -> ProviderInvocationResult<bool> {
+        let bucket_id = self.unalias(arg);
+        match self.s3_client.head_bucket().bucket(bucket_id).send().await {
+            Ok(_) => Ok(true),
+            Err(se) => match se.into_service_error() {
+                HeadBucketError::NotFound(_) => Ok(false),
+                e => {
+                    error!(error = ?e, "Unable to head bucket");
```

Comment:
Would you mind replacing these `error` fields with `err`? Unfortunately there's an issue somewhere in the tracing stack, where the `error` field gets set as a boolean and ignores whatever the client tried to set

Reply:
Ahh yes, will do!